src/odoo_sort_manifest_depends/sort_manifest_deps.py

```diff
@@ -53,17 +65,65 @@ def _identify_oca_addons(addon_names: list[str], odoo_series: OdooSeries) -> tup
                 distribution_name = addon_name_to_distribution_name(addon_name, odoo_series).replace("_", "-")
                 res = head(f"{OCA_ADDONS_INDEX_URL}{distribution_name}", timeout=REQUEST_TIMEOUT)
                 if res:
-                    category = "oca"
+                    category = get_oca_repository_name(addon_name, odoo_series) or oca_category_repo_not_found
                 else:
                     category = "other"
                 cache[addon_name] = category
 
-            if category == "oca":
-                oca_addons.append(addon_name)
-            else:
+            if category == "other":
                 other_addons.append(addon_name)
+            else:
+                oca_addons_by_category.setdefault(category, []).append(addon_name)
+
+    return oca_addons_by_category, other_addons
+
+
+def get_oca_repository_name(addon_name: str, odoo_series: OdooSeries) -> str | None:
+    specifier = SpecifierSet(f"=={odoo_series.value}.*")
+    distribution_name = addon_name_to_distribution_name(addon_name, odoo_series)
+    # get avaialble releases
+    project_url = simple.create_project_url(PYPI_SIMPLE_INDEX_URL, distribution_name)
+    response = get(project_url, headers={"Accept": simple.ACCEPT_JSON_V1}, timeout=REQUEST_TIMEOUT)
+    if response.status_code == PAGE_NOT_FOUND:
+        # project not found
+        return None
+    response.raise_for_status()
+    data = response.text
+    content_type = response.headers["Content-Type"]
+    project_details = simple.parse_project_details(data, content_type, distribution_name)
+    # find the first version that matches the requested Odoo version;
+    # we assume all releases come from the same repo for a given Odoo series
+    for file in project_details["files"]:
+        if file.get("yanked"):
+            continue
+        filename = file["filename"]
+        if not filename.endswith(".whl"):
+            continue
+        _, version, _, _ = parse_wheel_filename(filename)
+        if specifier.contains(version, prereleases=True):
+            # found a release that matches the requested Odoo version
+            break
+    else:
+        # no release found that matches the requested Odoo version
+        return None
+
+    pypi_json_url = _get_pypi_url(distribution_name, version)
+    response = get(pypi_json_url, timeout=REQUEST_TIMEOUT)
+    response.raise_for_status()
+    repo_url = response.json().get("info").get("home_page")
+
+    if not repo_url:
+        return None
+    return search("OCA/.*", repo_url).group()
```

Comment:
```suggestion
    return re.search("OCA/.*", repo_url).group()
```

When the function name is generic like that, I find it easier to have the module name here. I actually at to scroll up to understand what `search` was here.

Also take care that re.search can return None.